src/porepy/viz/exporter.py

```diff
@@ -520,546 +1064,789 @@ def _export_grid(
         num_pts = np.sum([g.num_nodes for g in gs])
         meshio_pts = np.empty((num_pts, 3))  # type: ignore
-        pts_pos = 0
 
-        # loop on all the 1d grids
+        # Initialize offset. Required taking into account multiple 1d grids.
+        nodes_offset = 0
+
+        # Loop over all 1d grids
         for g in gs:
-            # save the points information
-            sl = slice(pts_pos, pts_pos + g.num_nodes)
+
+            # Store node coordinates
+            sl = slice(nodes_offset, nodes_offset + g.num_nodes)
             meshio_pts[sl, :] = g.nodes.T
 
-            # Cell-node relations
-            g_cell_nodes = g.cell_nodes()
-            g_nodes_cells, g_cells, _ = sps.find(g_cell_nodes)
-            # Ensure ordering of the cells
-            g_nodes_cells = g_nodes_cells[np.argsort(g_cells)]
+            # Lines are simplices, and have a trivial connectvity.
+            cn_indices = self._simplex_cell_to_nodes(2, g)
 
-            # loop on all the grid cells
-            for c in np.arange(g.num_cells):
-                loc = slice(g_cell_nodes.indptr[c], g_cell_nodes.indptr[c + 1])
-                # get the local nodes and save them
-                cell_to_nodes[cell_type][cell_pos, :] = g_nodes_cells[loc] + pts_pos
-                cell_id[cell_type][cell_pos] = cell_pos
-                cell_pos += 1
+            # Add to previous connectivity information
+            cell_to_nodes[cell_type] = np.vstack(
+                (cell_to_nodes[cell_type], cn_indices + nodes_offset)
+            )
 
-            pts_pos += g.num_nodes
+            # Update offsets
+            nodes_offset += g.num_nodes
 
-        # construct the meshio data structure
-        num_block = len(cell_to_nodes)
-        meshio_cells = np.empty(num_block, dtype=object)
-        meshio_cell_id = np.empty(num_block, dtype=object)
+        # Construct the meshio data structure
+        num_blocks = len(cell_to_nodes)
+        meshio_cells = np.empty(num_blocks, dtype=object)
+        meshio_cell_id = np.empty(num_blocks, dtype=object)
 
+        # For each cell_type store the connectivity pattern cell_to_nodes for
+        # the corresponding cells with ids from cell_id.
         for block, (cell_type, cell_block) in enumerate(cell_to_nodes.items()):
             meshio_cells[block] = meshio.CellBlock(cell_type, cell_block.astype(int))
             meshio_cell_id[block] = np.array(cell_id[cell_type])
 
-        return meshio_pts, meshio_cells, meshio_cell_id
+        # Return final meshio data: points, cell (connectivity), cell ids
+        return Meshio_Geom(meshio_pts, meshio_cells, meshio_cell_id)
+
+    def _simplex_cell_to_nodes(
+        self, n: int, g: pp.Grid, cells: Optional[np.ndarray] = None
+    ) -> np.ndarray:
+        """
+        Determine cell to node connectivity for a general n-simplex mesh.
+
+        Parameters:
+            n (int): order of the simplices in the grid.
```

Comment:
"order" is what we usually refer to as "dimension", right? If so, could we use that description to avoid confusion (order could also be related to e.g. the orientation/node ordering in this context)